File: apollo/src/main/java/io/muun/apollo/data/preferences/SchemaVersionRepository.java
package io.muun.apollo.data.preferences;

import android.content.Context;
import com.f2prateek.rx.preferences.Preference;

import javax.inject.Inject;

public class SchemaVersionRepository extends BaseRepository {

    private static final String KEY_VERSION = "version";

    private final Preference<Integer> versionPreference;

    /**
     * Creates a repository.
     */
    @Inject
    public SchemaVersionRepository(Context context) {
        super(context);
        versionPreference = rxSharedPreferences.getInteger(KEY_VERSION);
    }

    @Override
    protected String getFileName() {
        return "schema_version";
    }

    /**
     * Returns the stored version, or 0 if none is stored.
     */
    public int getVersion() {
        return versionPreference.get();
    }

    public boolean hasVersion() {
        return versionPreference.isSet();
    }

    public void setVersion(int version) {
        versionPreference.set(version);
    }

}

File: android/apolloui/src/main/java/io/muun/apollo/data/preferences/SchemaVersionRepository.kt
package io.muun.apollo.data.preferences

import android.content.Context
import io.muun.apollo.data.preferences.rx.Preference
import javax.inject.Inject

class SchemaVersionRepository @Inject constructor(
    context: Context,
    repositoryRegistry: RepositoryRegistry,
) : BaseRepository(context, repositoryRegistry) {

    companion object {
        private const val KEY_VERSION = "version"
    }

    private val versionPreference: Preference<Int>
        get() = rxSharedPreferences.getInteger(KEY_VERSION)

    override val fileName get() = "schema_version"

    /**
     * Returns the stored version, or 0 if none is stored.
     */
    var version: Int
        get() = versionPreference.get()!!
        set(version) {
            versionPreference.set(version)
        }

    fun hasVersion(): Boolean {
        return versionPreference.isSet
    }
}